Source/CarthageKit/ProductExtension.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import Result
+import XCDBLD
+import ReactiveSwift
+
+/// Describes the type of packages file extensions
+public enum ProductExtension: String {
+	/// A .framework package.
+	case framework = "framework"
+	
+	/// A .action package.
+	case action = "action"
+	
+	/// A .bundle package.
+	case bundle = "bundle"
+	
+	/// A .kext package.
+	case kext = "kext"
+	
+	/// A .mdimporter package.
+	case mdimporter = "mdimporter"
+	
+	/// A .metallib package.
+	case metallib = "metallib"
+	
+	/// A .plugin package.
+	case plugin = "plugin"
+	
+	/// A .prefPane package.
+	case prefPane = "prefPane"
+	
+	/// A .qlgenerator package.
+	case qlgenerator = "qlgenerator"
+	
+	/// A .saver package.
+	case saver = "saver"
+	
+	/// A .xpc package.
+	case xpc = "xpc"
+	
+	public static let supportedExtensions: [ProductExtension] = [ .framework, .action, .bundle, .kext, .mdimporter, .metallib, .plugin, .prefPane, .qlgenerator, .saver, .xpc ]
```

Comment:
You could probably use `CaseIterable` instead?

Reply:
```suggestion
	public static let supportedExtensions: [ProductExtension] = ProductExtension.allCases
```